### ISSUE TITLE
508-defect-4 [AXE-CORE]: The Mapbox container SHOULD have role appropriate for interactive content

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#6103]
# [508-defect-4](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-4)

```diff
! Leaflet-Mapbox issue
```

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The Leaflet container **should** have a role appropriate for interactive content, as it is in the focus order. This is a custom widget, and should have corresponding valid ARIA role attribute values are applied, so the screen reader will properly announce it. In this case, `role="widget"` will allow the map to receive focus. Without this, the screen reader user won't know what this content is.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to know what content is interactive.

## Steps to Recreate

1. Enter `https://www.va.gov/find-locations` in browser
2. Start developer tools
3. Audit using axe-coconut
4. Verify the audit lists "Elements in the focus order need a role appropriate for interactive content" related to `.leaflet-container`
5. Return to the browser window with the Facility Locator
6. Start the screen reader device of your choice
7. Tab through the interface
8. When you get to the map, note that focus is allowed on the map but it is not announced as interactive

## Possible Fixes (optional)

To solve this violation, you need to:
Fix at least one (1) of these issues:
- Element does not have a widget role.
- Element has invalid semantics for an element in the focus order.

**Recommendation:** Add `role="widget"` to the `div.leaflet-container`.
```html
<div class="leaflet-container leaflet-touch leaflet-fade-anim leaflet-grab 
leaflet-touch-drag leaflet-touch-zoom" tabindex="0" 
style="min-height: 75vh; width: 100%; position: relative; outline: none;" role="widget">
```

## WCAG or Vendor Guidance (optional)

* [axe-core 3.4 - Elements in the focus order need a role appropriate for interactive content](https://dequeuniversity.com/rules/axe/3.4/focus-order-semantics?application=coconut)

**Comment by mmiddaugh** [comment]
I believe this issue can be closed, since we have moved away from the LEaflet version of Mapbox. Can you please confirm when you are able, @joshkimux or @1Copenut ?

**Comment by joshkimux** [comment]
@mmiddaugh I believe this issue still stands as mapbox would be considered as a widget too. Having the role would help screen reader users!
